src/redux/books/books.js

```diff
@@ -26,3 +37,48 @@ export const RemoveBook = (id) => ({
   type: REMOVE_BOOK,
   id,
 });
+
+export const addBookToServer = (book) => async (dispatch) => {
+  dispatch(AddBook(book));
+  const param = {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json; charset=UTF-8',
+    },
+    body: JSON.stringify({
+      item_id: book.id,
+      title: book.title,
+      author: book.author,
+      category: book.category,
+    }),
+
+  };
+  await fetch(`${baseUrl}books`, param);
+};
+
+export const getBooksFromServer = () => async (dispatch) => {
+  const response = await fetch(`${baseUrl}books`).then((response) => response.json());
+  const books = [];
+  const booksIds = Object.keys(response);
+  booksIds.map((key) => books.push({
+    id: key,
+    title: response[key][0].title,
+    author: response[key][0].author,
+    category: response[key][0].category,
+  }));
+  dispatch(SetBooks(books));
+};
```

Comment:
- [ ] Following the project requirements, you should use `createAsyncThunk` to create thunk action creators. Check out [this link](https://redux-toolkit.js.org/api/createAsyncThunk) to learn more about `createAsyncThunk`.

Reply:
I have tried implementing the `createAsyncThunk ` but it is not working as expected means it is posting a request to add book to server but it not rendering the added book to UI see here👇👇
```
export const addBookToServer = createAsyncThunk(
  'BOOK_STORE/BOOKS/ADD_BOOK',
  async (book) => {
    const param = {
      method: 'POST',
      headers: {
        'content-type': 'application/json; charset=UTF-8',
      },
      body: JSON.stringify({
        item_id: book.id,
        title: book.title,
        author: book.author,
        category: book.category,
      }),

    };
    await fetch(`${baseUrl}books`, param);
    return book;
  },
);
```
It would be nice if can take a look at code and help me figure out if i am missing something

Reply:
Hi @ishpaul777, sorry for the late reply. Anyway, well done with the "createAsyncThunk" implementation.